publish.go

```diff
@@ -0,0 +1,357 @@
+		matches := re.FindStringSubmatch(recentChangesPath)
+		if len(matches) == 2 {
+			language := matches[1]
+			content, err := fileutil.ReadStringFromFile(recentChangesPath)
+			if err != nil {
+				return map[string]string{}, err
+			}
+
+			recentChangesMap[language] = content
+		}
+	}
+	if len(recentChangesMap) > 0 {
+		log.Debugf("Found the following recent changes:")
+		for language, recentChanges := range recentChangesMap {
+			log.Debugf("%v\n", language)
+			log.Debugf("Content: %v", recentChanges)
+		}
+	} else {
+		log.Debugf("No recent changes found")
+	}
+
+	return recentChangesMap, nil
+}
+
+// getRelease gets a release from a track based on it's status. Note we would get error if we would like to have
+// multiple instances of a release with the same status. Example: "error: googleapi: Error 400: Too many completed
+// releases specified., releasesTooManyCompletedReleases".
+func getRelease(userFraction float64, releases *[]*androidpublisher.TrackRelease) *androidpublisher.TrackRelease {
+	status := releaseStatusFromConfig(userFraction)
+	for _, release := range *releases {
+		if status == release.Status {
+			return release
+		}
+	}
+	newRelease := androidpublisher.TrackRelease{
+		Status: status,
+	}
+	if userFraction != 0 {
+		newRelease.UserFraction = userFraction
+	}
+	*releases = append(*releases, &newRelease)
+	return &newRelease
+}
+
+// updateReleaseDetails creates and returns a new release object with the given version codes and adds the listing
+// information.
+func updateReleaseDetails(configs Configs, versionCodes googleapi.Int64s, release *androidpublisher.TrackRelease) error {
+	release.VersionCodes = versionCodes
+	log.Infof("Release version codes are: %v", release.VersionCodes)
+
+	log.Printf(" assigned app versions: %v", release.VersionCodes)
+	if err := updateListing(configs, release); err != nil {
+		return fmt.Errorf("failed to update listing, reason: %v", err)
+	}
+	return nil
+}
+
+// trackNamesToUpdate returns the tracks that should be updated.
+func trackNamesToUpdate(track string, tracks []*androidpublisher.Track) []string {
+	var possibleTrackNamesToUpdate []string
+	switch track {
+	case betaTrackName:
+		possibleTrackNamesToUpdate = []string{alphaTrackName}
+	case rolloutTrackName, productionTrackName:
+		possibleTrackNamesToUpdate = []string{alphaTrackName, betaTrackName}
+	}
+
+	var trackNamesToUpdate []string
+	for _, track := range tracks {
+		for _, trackNameToUpdate := range possibleTrackNamesToUpdate {
+			if trackNameToUpdate == track.Track {
+				trackNamesToUpdate = append(trackNamesToUpdate, trackNameToUpdate)
+			}
+		}
+	}
+
+	log.Printf(" possible tracks to update: %v", trackNamesToUpdate)
+	return trackNamesToUpdate
+}
+
+// unTrackFromTracks untracks the lower level versions.
+func unTrackFromTracks(trackNamesToUpdate []string, versionCodes googleapi.Int64s, service *androidpublisher.Service, packageName string, appEditID string) error {
```

Comment:
Could call untrackFromTracks, untrack as on word (if it is).

Reply:
Renamed.